frontend/src/dashboard.js

```diff
@@ -46,67 +46,84 @@ import {
   ResultAggregatorWidget,
   ResultSummaryWidget
 } from './widgets';
-import { getActiveProject, getActiveDashboard } from './utilities.js';
+import { IbutsuContext } from './services/context.js';
 
 
 export class Dashboard extends React.Component {
+  static contextType = IbutsuContext;
   static propTypes = {
-    eventEmitter: PropTypes.object
+    eventEmitter: PropTypes.object,
+    navigate: PropTypes.func,
+    params: PropTypes.object,
   }
 
   constructor(props) {
     super(props);
-    let dashboard = getActiveDashboard() || this.getDefaultDashboard();
     this.state = {
       widgets: [],
       filteredDashboards: [],
       dashboards: [],
-      selectedDashboard: dashboard,
+      selectedDashboard: null,
       isDashboardSelectorOpen: false,
       isNewDashboardOpen: false,
       isWidgetWizardOpen: false,
       isEditModalOpen: false,
       editWidgetData: {},
-      dashboardInputValue: dashboard?.title || '',
+      dashboardInputValue: '',
       filterValueDashboard: ''
     };
-    props.eventEmitter.on('projectChange', () => {
-      this.clearDashboards();
-      this.getDashboards();
+    props.eventEmitter.on('projectChange', (value) => {
+      this.getDashboards(value);
+      this.getDefaultDashboard(value);
     });
   }
 
-  getDefaultDashboard() {
-    let project = getActiveProject();
-    if (project && project.defaultDashboard) {
-      console
-      const dashboard = JSON.stringify(project.defaultDashboard)
-      localStorage.setItem('dashboard', dashboard);
-      return project.defaultDashboard;
+  sync_context = () => {
+    // Active dashboard
+    const { activeDashboard } = this.context;
+    const { selectedDashboard } = this.state;
+    const paramDash = this.props.params?.dashboard_id;
+    let updatedDash = undefined;
+    // API call to update context
+    if ( paramDash && activeDashboard?.id !== paramDash) {
```

Comment:
I was able to catch an issue here when I run a fresh ibutsu locally and had only projects created. `const paramDash = this.props.params?.dashboard_id;` returned me a null but as a string. So if didn't catch this and this is what I got:

![image](https://github.com/user-attachments/assets/233d5f75-156f-4b13-8b08-1d987e32641a)